### PR TITLE
Some AVM1 fixes

### PR DESCRIPTION
**Pull request by tschneidereit**
Makes SWF 123 from the archive show more stuff. Still not fully working, but progress.

r? @yurydelendik.

### CODE REVIEW

File: src/avm1/lib/AVM1MovieClip.ts [hunk 1]
```diff
@@ -343,7 +343,7 @@ module Shumway.AVM1.Lib {
       var nativeObject = this.as3Object;
       var maxDepth = 0;
       for (var i = 0, numChildren = nativeObject.numChildren; i < numChildren; i++) {
-        var child = nativeObject._lookupChildByIndex(i);
+        var child = nativeObject._children[i];
```
Comment (yurydelendik) [inline src/avm1/lib/AVM1MovieClip.ts]: I'm okay with that, but I would rather add a LookupChildOptions parameter to the _lookupChildByIndex (see _lookupChildByName)

Comment (tschneidereit) [inline src/avm1/lib/AVM1MovieClip.ts]: > I'm okay with that, but I would rather add a LookupChildOptions parameter to the _lookupChildByIndex (see _lookupChildByName)

Ok, changed in the new version. Again, r?

**Comment by yurydelendik** [comment]
r+

**Pull request merged**